dotCMS/src/main/java/com/dotmarketing/db/DbConnectionFactory.java

```diff
@@ -100,20 +112,34 @@ public static DataSource getDataSource() {
             synchronized (DbConnectionFactory.class) {
 
                 if (null == defaultDataSource) {
+                    if (DBPropertiesDatasourceStrategy.getInstance().existsDBPropertiesFile()) {
+                        defaultDataSource = DBPropertiesDatasourceStrategy.getInstance().getDatasource();
+                        Logger.info(DbConnectionFactory.class, "Datasource loaded from db.properties file");
+                    } else if (System.getenv("connection_db_base_url") != null) {
+                        defaultDataSource = SystemEnvDatasourceStrategy.getInstance().getDatasource();
+                        Logger.info(DbConnectionFactory.class, "Datasource loaded from system environment");
+                    } else {
+                        defaultDataSource = DockerSecretDatasourceStrategy.getInstance().getDatasource();
+                        Logger.info(DbConnectionFactory.class, "Datasource loaded from Docker Secret");
+                    }
+
+                    if (null == defaultDataSource){
+                        defaultDataSource = TomcatDatasourceStrategy.getInstance().getDatasource();
+                        Logger.info(DbConnectionFactory.class, "Datasource loaded from context.xml");
+                    }
```

Comment:
test?

Reply:
It's difficult to add a test to this method without using PowerMockito